blocks/library/embed/index.js

```diff
@@ -110,20 +140,41 @@ function getEmbedBlockSettings( { title, icon, category = 'embed', transforms, k
 				const { url } = this.props.attributes;
 				const { setAttributes } = this.props;
 
+				// If we don't have any URL patterns, or we do and the URL doesn't match,
+				// then we should look for a block that has a matching URL pattern.
+				if ( ! patterns || ( patterns && ! matchesPatterns( url, patterns ) ) ) {
+					const matchingBlock = findBlock( url );
+					// WordPress blocks can work on multiple sites, and so don't have patterns.
+					if ( 'core-embed/wordpress' !== this.props.name && 'core/embed' !== matchingBlock ) {
+						// At this point, we have discovered a more suitable block for this url, so transform it.
+						if ( this.props.name !== matchingBlock ) {
+							this.props.onReplace( createBlock( matchingBlock, { url } ) );
+							return;
+						}
+					}
+				}
+
 				this.setState( { error: false, fetching: true } );
-				wp.apiRequest( { path: `/oembed/1.0/proxy?${ stringify( { url } ) }` } )
+				wpEmbedAPI( url )
 					.then(
 						( obj ) => {
 							if ( this.unmounting ) {
 								return;
 							}
-
-							const { html, provider_name: providerName } = obj;
-							const providerNameSlug = kebabCase( toLower( providerName ) );
-							let { type } = obj;
+							// Some plugins put the embed html in `result`, so get the right one here.
+							const html = obj.html ? obj.html : obj.result;
+							// Block titles are the provider names, but we can override with `providerName` too.
+							const providerNameSlug = kebabCase( toLower( '' !== providerName ? providerName : title ) );
```

Comment:
Why don't we take the providerName from the oembed response any more?

Because there are plugins that bypass oembed completely, and don't provide it. And the titles of the blocks are the provider names anyway.